label_studio/io_storages/base_models.py

```diff
@@ -74,12 +74,30 @@ def _scan_and_create_links(self, link_class):
                 continue
             logger.debug(f'{self}: found new key {key}')
             data = self.get_data(key)
+
+            # predictions
+            predictions = data.get('predictions', [])
+            if predictions:
+                if 'data' not in data:
+                    raise ValueError('If you use "predictions" field in the task, '
+                                     'you must put "data" field in the task too')
+                data = data['data']
+
             with transaction.atomic():
                 task = Task.objects.create(data=data, project=self.project)
                 link_class.create(task, key, self)
                 logger.debug(f'Create {self.__class__.__name__} link with key={key} for task={task}')
                 tasks_created += 1
 
+                if not task.predictions.exists():
```

Comment:
if the task exists already and this is the second sync, predictions will be duplicated without this constraint

Reply:
but it seems that duplication is handled by https://github.com/heartexlabs/label-studio/blob/365dc78a50d7d029c2bb6087def4641f94bf4fd5/label_studio/io_storages/base_models.py#L72 otherwise even task creation won't work